homepage/homepage/content/docs/server-side/http-requests.mdx

```diff
@@ -0,0 +1,227 @@
+import { JazzRequestError } from "jazz-tools";
+
+export async function POST(request: Request) {
+  return bookEventTicket.handle(
+    request,
+    jazzServer.worker,
+    async ({ event }, madeBy) => {
+      // Check if the event is full
+      if (event.reservations.length >= event.capacity) {
+        // The JazzRequestError is propagated to the client, use it for any validation errors
+        throw new JazzRequestError("Event is full", 400);
+      }
+
+      const ticketGroup = Group.create(jazzServer.worker);
+      const ticket = Ticket.create({
+        account: madeBy,
+        event,
+      });
+
+      // Give access to the ticket to the client
+      ticketGroup.addMember(madeBy, "reader");
+
+      event.reservations.push(ticket);
+
+      return {
+        ticket,
+      };
+    },
+  );
+}
+```
+</CodeGroup>
+
+<Alert variant="info" className="mt-4">
+  **Note**: To ensure that the limit is correctly enforced, the handler should be deployed in a single worker instance (e.g. a single Cloudflare DurableObject).
+  
+  Details on how to deploy a single instance Worker are available in the [Deployments & Transactionality](#deployments--transactionality) section.
+</Alert>
+
+### Client-side error handling
+
+Handle errors on the client side:
+
+<CodeGroup>
+```ts
+import { bookEventTicket, Ticket, Event } from "@/schema";
+import { isJazzRequestError } from "jazz-tools";
+
+export async function sendEventBookingRequest(event: Event) {
+  try {
+    const { ticket } = await bookEventTicket.send({ event });
+
+    return ticket;
+  } catch (error) {
+    // This works as a type guard, so you can easily get the error message and details
+    if (isJazzRequestError(error)) {
+      alert(error.message);
+      return;
+    }
+  }
+}
+```
+</CodeGroup>
+
+<Alert variant="info" className="mt-4">
+  **Note**: The `experimental_defineRequest` API is still experimental and may change in future versions. For production applications, consider the stability implications.
+</Alert>
+
+## Deployments & Transactionality
+
+### Single Instance Requirements
+
+Deploy your Server Worker on a single instance when applying validation rules that require atomic operations.
+
+For example, when implementing capacity limits in an event registration system, you need a single instance to ensure the capacity check and ticket creation happen atomically.
+
+Jazz is an eventually consistent database, and with multiple region deployments you may accept multiple registrations at the same time
+and realize after the sync that the event is over capacity.
+
+Transactional updates on multi-region deployments are part of the roadmap, but not yet supported.
+
+Examples of when you must deploy on a single instance are:
+- when a counter is involved
+- when there is any kind of limit on the resources (like capacity)
+
+Essentially when the output of the request depends on the state of the database, you must deploy on a single instance.
```

Comment:
I like having a simple heuristic, but I feel this one is too heavy handed.

E.g. if I want to aggregate data across a lot of users to display some metrics I can have multiple server instances, because I'm not looking to provide _exact_ information.

Single servers are really only necessary to enforce invariants or provide a consistent view of the data, right? Maybe we can modify the heuristic to be along those lines (probably using more user-friendly terms than _invariants_ and _consistent_ 😅)